bigbluebutton-html5/imports/ui/components/modal/random-user/container.jsx

```diff
@@ -11,96 +10,143 @@ import logger from '/imports/startup/client/logger';
 
 const SELECT_RANDOM_USER_ENABLED = Meteor.settings.public.selectRandomUser.enabled;
 
+const SELECT_RANDOM_USER_COUNTDOWN = Meteor.settings.public.selectRandomUser.countdown;
+
+//  Time intervals in milliseconds
+//  for iteration in animation.
+const intervals = [0, 200, 450, 750, 1100, 1500];
+
 //  A value that is used by component to remember
 //  whether it should be open or closed after a render
 let keepModalOpen = true;
 
+const toggleKeepModalOpen = () => { keepModalOpen = !keepModalOpen; };
+
 //  A value that stores the previous indicator
-let updateIndicator = 1;
+let prevUpdateIndicator = true;
 
-const toggleKeepModalOpen = () => { keepModalOpen = ! keepModalOpen; };
+let restartLimit = 5;
 
 const RandomUserSelectContainer = (props) => {
   const usingUsersContext = useContext(UsersContext);
   const { users } = usingUsersContext;
-  const { randomlySelectedUser } = props;
+  let { randomlySelectedUser } = props;
+
+  let viewerPool = [];
+
+  // Will contain an array with 6 values,
+  // each contaiing an ID and delay in ms for animation
+  // first 5 for animation and fifth chosen in akka-apps
+  let userList = [];
 
+  // Will contain visual data about those 6 users
   let mappedRandomlySelectedUsers = [];
 
   const currentUser = {
     userId: Auth.userID,
     presenter: users[Auth.meetingID][Auth.userID].presenter
   };
 
-  try {
-    if (!currentUser.presenter //  this functionality does not bother presenter
-      && (!keepModalOpen) //  we only ween a change if modal has been closed before
-      && (randomlySelectedUser[0][1] !== updateIndicator)// if tey are different, a user was generated
-    ) { keepModalOpen = true; } //  reopen modal
-    if (!currentUser.presenter) { updateIndicator = randomlySelectedUser[0][1]; } // keep indicator up to date
-  } catch (err) {
-    logger.error({
-      logCode: 'Random_USer_Error',
-      extraInfo: {
-        stackTrace: err,
-      },
-    },
-    '\nIssue in Random User Select container caused by back-end crash'
-      + '\nValue of 6 randomly selected users was passed as '
-      + `{${randomlySelectedUser}}`
-      + '\nHowever, it is handled.'
-      + '\nError message:'
-      + `\n${err}`);
+  // Populating viewerPool
+  for (let meeting in users) {
+    for (let id in users[meeting]) {
+      if(users[meeting][id].role == 'VIEWER') viewerPool.push(id);
```

Comment:
Looks like you accidentally dropped the third `=`. Let's use `===` instead of `!==`.